src/modules/auth/index.js

```diff
@@ -191,31 +190,58 @@ export default class AuthModule extends Module {
 		});
 	}
 
-
-	async _updateUserInfo() {
+	async _updateUserInfo(that = this, oldUserInfo = null, fAlert = false) {
 		let response;
 		try {
-			response = await this.Api.userinfo(this.OAuthToken.access_token);
+			response = await that.Api.userinfo(that.OAuthToken.access_token);
 		}
 		catch (err) {
 			console.log("Failed to update user info", err);
-			this.UserInfo = null;
-			if (this.App.Store != null) {
-				this.App.Store.dispatch({ type: types.AUTH_USERINFO, payload: this.UserInfo });
+			// If session has expired
+			if (oldUserInfo) {
+				oldUserInfo = null;
+				that.App.addAlert("warning", "Your session has expired");
+			}
+			that.UserInfo = null;
+			if (that.App.Store != null) {
+				that.App.Store.dispatch({ type: types.AUTH_USERINFO, payload: that.UserInfo });
 			}
 			return false;
 		}
 
-		this.UserInfo = response.data;
-		if (this.App.Store != null) {
-			this.App.Store.dispatch({ type: types.AUTH_USERINFO, payload: this.UserInfo });
+		that.UserInfo = response.data;
+		if (that.App.Store != null) {
+			that.App.Store.dispatch({ type: types.AUTH_USERINFO, payload: that.UserInfo });
 		}
 
 		// Check for TenantService and pass tenants list obtained from userinfo
-		let tenants_list = this.UserInfo.tenants;
-		if (this.App.Services.TenantService) {
-			this.App.Services.TenantService.set_tenants(tenants_list);
+		let tenants_list = that.UserInfo.tenants;
+		if (that.App.Services.TenantService) {
+			that.App.Services.TenantService.set_tenants(tenants_list);
+		}
+
+		/* Notifying user about session expiration section */
+		let exp = new Date(that.UserInfo.exp).getTime(); // Expiration timestamp
+		const difference = exp - Date.now(); // Difference between current time and expiration date in milliseconds
+
+		// If userinfo expiration date has been changed then first alert flag
```

Comment:
@titkov49 Can you wrap multiple line comments into `/* Your comment */` and do not comment every each line with `// Your comment` ? Just for sake of some structural styling